src/components/implementation/tests/micro_booter/mb_tests.c

```diff
@@ -332,69 +157,125 @@ spinner(void *d)
 		;
 }
 
+void
+clear_sched(int* rcvd, thdid_t* tid, int* blocked, cycles_t* cycles, tcap_time_t* thd_timeout)
+{
+	while (cos_sched_rcv(BOOT_CAPTBL_SELF_INITRCV_CPU_BASE, RCV_ALL_PENDING, 0,
+			     rcvd, tid, blocked, cycles, thd_timeout) != 0)
+		;
+}
+
+#define NUM_TEST 16 			/* Iterator NUM */
+#define MAX_THRS 4  			/* Max Threshold Multiplier */
+#define MIN_THRS 1  			/* Min Threshold Multiplier */
+#define GRANULARITY 1000 		/* Timer Granularity*/
+#define T_ITER 16
+
 static void
 test_timer(void)
 {
-	int      i;
+	int	 i;
 	thdcap_t tc;
 	cycles_t c = 0, p = 0, t = 0;
 
-	PRINTC("Starting timer test.\n");
 	tc = cos_thd_alloc(&booter_info, booter_info.comp_cap, spinner, NULL);
 
-	for (i = 0; i <= 16; i++) {
+	for (i = 0; i <= T_ITER; i++)
+	{
 		thdid_t     tid;
-		int         blocked, rcvd;
+		int	    blocked, rcvd;
 		cycles_t    cycles, now;
 		tcap_time_t timer, thd_timeout;
 
 		rdtscll(now);
-		timer = tcap_cyc2time(now + 1000 * cyc_per_usec);
+		timer = tcap_cyc2time(now + GRANULARITY * cyc_per_usec);
 		cos_switch(tc, BOOT_CAPTBL_SELF_INITTCAP_CPU_BASE, 0, timer, BOOT_CAPTBL_SELF_INITRCV_CPU_BASE,
-		           cos_sched_sync());
+			   cos_sched_sync());
 		p = c;
 		rdtscll(c);
-		if (i > 0) t += c - p;
+		if (i > 0) {
+			t += c - p;
+		}
 
-		while (cos_sched_rcv(BOOT_CAPTBL_SELF_INITRCV_CPU_BASE, RCV_ALL_PENDING, 0,
-				     &rcvd, &tid, &blocked, &cycles, &thd_timeout) != 0)
-			;
+		clear_sched(&rcvd, &tid, &blocked, &cycles, &thd_timeout);
 	}
+	EXPECT_LLU_LT((long long unsigned)(t/T_ITER), (unsigned)(GRANULARITY * cyc_per_usec * MAX_THRS), "Test Timer MAX");
+	EXPECT_LLU_LT((unsigned)(GRANULARITY * cyc_per_usec * MIN_THRS), (long long unsigned)(t/T_ITER), "Test Timer MIN");
 
-	PRINTC("\tCycles per tick (1000 microseconds) = %lld, cycles threshold = %u\n", t / 16,
-	       (unsigned int)cos_hw_cycles_thresh(BOOT_CAPTBL_SELF_INITHW_BASE));
+	/* TIMER IN PAST */
```

Comment:
declarations and body mixed.

Reply:
`error: redeclaration of ‘blocked’ with no linkage`

When I declare everything together.